test/java/clojure/lang/UtilTest.java

```diff
@@ -0,0 +1,227 @@
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import java.math.BigInteger;
+import java.util.HashMap;
+
+public class UtilTest {
+
+  @Rule public final ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void testClassOf() {
+    Assert.assertNull(Util.classOf(null));
+    Assert.assertEquals(Util.classOf(0), Integer.class);
+  }
+
+  @Test
+  public void testCompare() {
+    Assert.assertEquals(1, Util.compare(false, null));
+    Assert.assertEquals(0, Util.compare(1, 1));
+    Assert.assertEquals(-1, Util.compare(null, 0));
+    Assert.assertEquals(
+            0, Util.compare(new Integer(1), new Integer(1)));
+  }
+
+  @Test
+  public void testEquals() {
+    Assert.assertTrue(Util.equals(1, 1));
+    Assert.assertFalse(Util.equals(1, 0));
+    Assert.assertFalse(Util.equals(null, 0));
+  }
+
+  @Test
+  public void testEquivWithLong() {
+    Assert.assertTrue(Util.equiv((Object) 0, 0L));
+    Assert.assertFalse(Util.equiv(0L, (Object) 1));
+    Assert.assertFalse(Util.equiv(1L, 0L));
+    Assert.assertTrue(Util.equiv(1L, 1L));
+  }
+
+  @Test
+  public void testEquivWithDouble() {
+    Assert.assertTrue(Util.equiv((Object) 0.0, 0.0));
+    Assert.assertFalse(Util.equiv(0.0, (Object) 1.0));
+    Assert.assertFalse(Util.equiv(1.0, 0.0));
+    Assert.assertTrue(Util.equiv(1.0, 1.0));
+  }
+
+  @Test
+  public void testEquivWithBoolean() {
+    Assert.assertFalse(Util.equiv(false,(Object) 0));
+    Assert.assertFalse(Util.equiv((Object)0, false));
+    Assert.assertFalse(Util.equiv(true, false));
+    Assert.assertTrue(Util.equiv(false, false));
+  }
+
+  @Test
+  public void testEquivWithChar() {
+    Assert.assertFalse(Util.equiv('\u0001', '\u0000'));
+    Assert.assertTrue(Util.equiv('\u0000', '\u0000'));
+    Assert.assertFalse(Util.equiv(null, '\u0000'));
+    Assert.assertFalse(Util.equiv('\u0000', null));
+  }
+
+  @Test
+  public void testEquivWithObject() {
+    Assert.assertTrue(Util.equiv(null, null));
+    Assert.assertFalse(Util.equiv(null, new Integer(1)));
+    Assert.assertTrue(Util.equiv(new Integer(1), new Integer(1)));
+  }
+
+  @Test
+  public void testEquivPredColl1() {
+    final HashMap k1 = new HashMap();
+    k1.put(null, null);
+    final HashMap k2 = new HashMap();
+
+    Assert.assertFalse(Util.equivPred(k1).equiv(k1, k2));
+
+    k2.put(null, null);
+
+    Assert.assertTrue(Util.equivPred(k1).equiv(k1, k2));
+  }
+
+  @Test
+  public void testEquivPredColl2() {
+    ArraySeq k1 = new ArraySeq(new String[1], 0);
+    ArraySeq k2 = new ArraySeq(new String[1], 0);
+
+    Assert.assertTrue(Util.equivPred(k1).equiv(k1, k2));
+
+    k2 = new ArraySeq(new String[2], 0);
+
+    Assert.assertFalse(Util.equivPred(k1).equiv(k1, k2));
+  }
+
+
+  @Test
+  public void testEquivPredNull() {
+    Object k1 = null;
+    Object k2 = null;
+
+    Assert.assertTrue(Util.equivPred(k1).equiv(k1, k2));
```

Comment:
Can you not do ` Assert.assertTrue(Util.equivPred((object)null).equiv((object)null, (object)null));` ?

Reply:
Yes you can and this is how I originally refactored it but then it is very unclear that the first 2 inputs are supposed to be the exact same thing